api/src/test/java/org/openmrs/api/ObsServiceTest.java

```diff
@@ -812,6 +813,46 @@ public void getObs_shouldGetObsMatchingGivenObsId() {
 		assertEquals(5089, obs.getConcept().getId().intValue());
 	}
 	
+	/**
+	 * This is a test to verify the behavior reported in TRUNK-5721.
+	 * The test name and the content will be refactored once the expected behavior is known
```

Comment:
What do you mean by once the expected behaviour is known?

Reply:
I’d like to confirm whether a complx observation with `ComplexData#data` set to null, should be saved to the database?

If it is okay to be saved, then I could make the changes, so `ObsService#getObs` would allow retrieving complex observation with no data files.

Reply:
I would not find it correct to save complex data without data. 😊

Reply:
Ok, thank you.. I'll also make a community post, if there's more things to clarify.